### PR TITLE
feat: log client info in requests

### PR DESCRIPTION
**Pull request by PierreLeveau**
This adds the client info (name, version, language, method) into the request headers. 

this gives:
image

### CODE REVIEW

File: src/kili/graphql/graphql_client.py [hunk 1]
```diff
@@ -137,8 +155,19 @@ def execute(self, query: Union[str, DocumentNode], variables: Optional[Dict] = N
             variables: the payload of the query
         """
         document = query if isinstance(query, DocumentNode) else gql(query)
+
         try:
-            result = self._gql_client.execute(document=document, variable_values=variables)
+            assert self._gql_transport.headers
```
[the remark this answers is not in the document]
Comment (PierreLeveau) [reply src/kili/graphql/graphql_client.py:160]: self._gql_transport.headers can be None
so I'm adding a message.

**Review by Jonas1312 — APPROVED** [review]

File: tests/graphql/test_graphql_client.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import mock
```
Comment (theodu) [inline tests/graphql/test_graphql_client.py:1]: there were already a file tests/test_graphql_client.py 
Can we add the test inside in this new file too ?

File: tests/graphql/test_graphql_client.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        verify=True,
+        retries=20,
+        method="POST",
+    )
+    graphql_client.included_logged_method_locations = ["test_graphql_client"]
+    graphql_client.execute(query="dummy", variables={})
+
+    assert graphql_client._gql_client.execute.call_args[1]["extra_args"] == {
+        "headers": {"kili-client-method-name": None}
+    }
```
Comment (theodu) [inline tests/graphql/test_graphql_client.py:47]: don't we add a test to test that we have the right kili-client-method-name called ?

**Review by theodu — CHANGES_REQUESTED** [review]

File: src/kili/utils/logcontext.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+from datetime import datetime
+from typing import Callable, Dict, List
+
+from kili import __version__
+from kili.graphql.clientnames import GraphQLClientName
+
+
+class Singleton(type):
+    """
+    Utitility meta-class to guarantee single instance
```
Comment (Jonas1312) [inline src/kili/utils/logcontext.py:16]: typo

**Review by theodu — APPROVED** [review]
very impressive!

**Review by Jonas1312 — APPROVED** [review]
I tested it, works great

**Pull request merged**